### PR TITLE
Create "row click" shortcuts for a few widgets

### LINKED ISSUE #90: make switch touch targets full-screen
For simple binary switches, the "button" can actually be the full width and height of the cell containing the icon, text, and button.
##

### LINKED ISSUE #550: Setpoint and Selection dialogs
My app just updated and now the Selection and Setpoint Sitemap items both open a dialog, instead of their old behavior. This makes a bit more sense for the Selection item than it does for a Setpoint, but I don't like this new dialog instead of the nicely sized dropdown list that we had before. Setpoints are my big issue here, why have they been changed to a wheel dialog? This splits the user experience between android and the web. The web interface still uses the +/- buttons, as is indicated in all of the documentation.

### LINKED ISSUE #735: Support Slider "switchSupport"
It's pretty common for home automation software to support having both a switch and a slider-like element combined into one for dimmable lights – Z-Way, ImperiHome, Fibaro Home Center, Vera etc. Slider elements in openHab already have the "switchSupport" parameter, but this is only used in Classic UI, and there only to modify the function of the already existing up and down buttons. In the Android UI one is left with having to create two elements for the same thing, one Switch and one Slider, with basically the same label and controlling the same thing, taking up twice the real estate and making the control slightly more awkward than it needs to be.

I would love to see switchSupport implemented in the Android app as well, but rendered as a single element with both a switch and a slider bar, much like this (although this image is from manipulating a Basic UI element, you should get the idea):

switchSupport rendered

The usefulness of this is e.g. for Z-Wave items that have support for ON and OFF commands as well as for precise levels. Usually ON has a value of 255, and will often turn the device back on to its previous dimmer level (or a specific preset value, or something else entirely, all dependent on the physical device and its configuration). So this must not be confused with ON just being a shortcut for setting the slider to 100% – something on which the documentation is a bit unclear with regards to Classic UI functionality.

### PR DESCRIPTION
**Pull request by mueller-ma**
Closes #90
Closes #550
Closes #735 

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java [hunk 1]
```diff
@@ -579,6 +588,12 @@ public void bind(Widget widget) {
             }
         }
 
+        @Override
+        protected void handleRowClick() {
+            setSeekBarProgress(mSeekBar.getProgress() == mBoundWidget.minValue()
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java]: This is where #735 can be fixed. I'll implement it.

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java [hunk 2]
```diff
@@ -579,6 +588,17 @@ public void bind(Widget widget) {
             }
         }
 
+        @Override
+        protected void handleRowClick() {
+            if (mBoundWidget.switchSupport() && mBoundWidget.item() != null) {
+                Util.sendItemCommand(mConnection.getAsyncHttpClient(), mBoundWidget.item(),
+                        mSeekBar.getProgress() == mBoundWidget.minValue() ? "ON" : "OFF");
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java]: Widget min value equals progress 0, see lines 584/585.

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java [hunk 3]
```diff
@@ -579,6 +588,17 @@ public void bind(Widget widget) {
...
+            } else {
+                setSeekBarProgress(mSeekBar.getProgress() == mBoundWidget.minValue()
+                        ? mBoundWidget.maxValue() : mBoundWidget.minValue());
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java]: If a user has switchSupport enabled for some sliders, he might find it obscure if nothing happens for other sliders on row click.
Maybe there is a slider used in an AOSP or Google product that we can use as reference.

[the remark this answers is not in the document]
Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java]: If there is, go for it ;-) Personally I haven't seen that UI pattern (toggle slider position on row click) so far.

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java (hunk 2 above)
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java]: Fixed.

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.java]: I haven't seen this in any app so far, removed.

**Pull request merged**